modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -305,6 +314,10 @@ The function imreadmulti loads a specified range from a multi-page image from th
 */
 CV_EXPORTS_W bool imreadmulti(const String& filename, CV_OUT std::vector<Mat>& mats, int start, int count, int flags = IMREAD_ANYCOLOR);
 
+CV_EXPORTS_W bool imreadanimation(const String& filename, CV_OUT Animation& animation);
```

Comment:
Animation is a struct:
Pros:
- easy to bind
- all info in one spot

Cons:
- decoding everything at once can suck up a lot of RAM so maybe a VideoReader-like API would be better

I am not sure of the best way to go there. @asmorkalov , any quick opinion?

Reply:
Let's stick to struct, they are easier to bind. Animations do not have 100 frames.